lib/ethpy/ethpy/hyperdrive/api/_contract_calls.py

```diff
@@ -377,14 +390,16 @@ async def _async_close_short(
     # before calling smart contract transact
     # Since current_pool_state.block_number is a property, we want to get the static block here
     current_block = interface.current_pool_state.block_number
-    preview_result = smart_contract_preview_transaction(
-        interface.hyperdrive_contract,
-        agent_checksum_address,
-        "closeShort",
-        *fn_args,
-        block_number=current_block,
-        read_retry_count=interface.read_retry_count,
-    )
+    preview_result = {}
+    if preview_before_trade or slippage_tolerance is not None:
+        preview_result = smart_contract_preview_transaction(
+            interface.hyperdrive_contract,
+            agent_checksum_address,
+            "closeShort",
+            *fn_args,
+            block_number=current_block,
+            read_retry_count=interface.read_retry_count,
+        )
     if slippage_tolerance:
```

Comment:
should this be `slippage_tolerance is not None` to match the other patterns?

Reply:
ya I like it more explicit